docs/LED_Effect.md

```diff
@@ -108,38 +108,74 @@ layers:
+particular effect again.
 To deactivate all effects we can use the GCode command `STOP_LED_EFFECTS`.
 To only deactivate effects for certain LEDs we can specify the LEDS parameter:
-`STOP_LED_EFFECTS LEDS="neopixel:panel_ring"` You can also specify indeces (see 
-below): `STOP_LED_EFFECTS LEDS="neopixel:panel_ring (1-7)"`. Only one 
-LED parameter can be specified at a time. To stop the effects for multiple LEDs 
+`STOP_LED_EFFECTS LEDS="neopixel:panel_ring"` You can also specify indeces (see
+below): `STOP_LED_EFFECTS LEDS="neopixel:panel_ring (1-7)"`. Only one
+LED parameter can be specified at a time. To stop the effects for multiple LEDs
 we have to run the command multiple times.
 
 #### Fading in and out
 Effects can be faded in and out by specifying the `FADETIME` parameter:
-`SET_LED_EFFECT EFFECT=panel_idle FADETIME=1.0` fades the effect in during one 
-second. Running `SET_LED_EFFECT EFFECT=panel_idle STOP=1 FADETIME=1.0` fades it 
-out in one second. We can also fade out all effects by running 
-`STOP_LED_EFFECTS FADETIME=1.0`. It is also possible to crossfade effects by 
-using the `REPLACE` parameter with `SET_LED_EFFECT` (see above): 
+`SET_LED_EFFECT EFFECT=panel_idle FADETIME=1.0` fades the effect in during one
+second. Running `SET_LED_EFFECT EFFECT=panel_idle STOP=1 FADETIME=1.0` fades it
+out in one second. We can also fade out all effects by running
+`STOP_LED_EFFECTS FADETIME=1.0`. It is also possible to crossfade effects by
+using the `REPLACE` parameter with `SET_LED_EFFECT` (see above):
 `SET_LED_EFFECT EFFECT=panel_idle REPLACE=1 FADETIME=1.0`
 
 #### Restarting Effects
-When an effect is stopped and then started again, it resumes from the frame where 
-it last left off.  To restart the effect from the beginning, specify the `RESTART` 
+When an effect is stopped and then started again, it resumes from the frame where
+it last left off.  To restart the effect from the beginning, specify the `RESTART`
 parameter: `SET_LED_EFFECT EFFECT=panel_idle RESTART=1`.
 
+#### Template processing
+The effect layers are processed as templates. That means that they can contain
+the same control logic that Klipper macros do. However, processing layers can
+be computationally intensive, which may affect the performance of the Raspberry
+Pi host. Therefore, layers are only evaluated as templates on effect creation
+(when the config is read in).
+
+If there is a need to evaluate the layers everytime the effect is activated,
+set the `dynamic` effect-level parameter to `true`.
+
+Setting the `dynamic` setting to `true` will also allow the effect to accept
+parameters just like `gcode_macro`. Parameters can be passed through the
+`SET_LED_EFFECT` command.
+
+The following is an example of a layer that accepts parameters:
+
+```
+[led_effect param_effect]
+autostart: false
+dynamic: true
+leds:
+    neopixel:leds
+layers:
+    blink {params.DURATION|default(1)|int} {params.CYCLE|default(0.5)|float} top (1.0, 0,0, 0,0)
```

Comment:
Also you shouldn't convert the duration parameter to int, but also to float.

Reply:
Done.